staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go

```diff
@@ -194,11 +255,14 @@ func (a *ApplySet) FetchParent() error {
 	helper := resource.NewHelper(a.client, a.parentRef.RESTMapping)
 	obj, err := helper.Get(a.parentRef.Namespace, a.parentRef.Name)
 	if errors.IsNotFound(err) {
+		if !builtinApplySetParentGVRs.Has(a.parentRef.Resource) {
+			return fmt.Errorf("custom resource ApplySet parents cannot be created automatically")
+		}
 		return nil
 	} else if err != nil {
-		return fmt.Errorf("failed to fetch ApplySet parent object %q from server: %w", a.parentRef, err)
+		return fmt.Errorf("failed to fetch ApplySet parent object %q: %w", a.parentRef, err)
```

Comment:
In reality, this always returns an error, but  that error does not come from the server:

```
I0307 23:44:07.723505   50417 round_trippers.go:553] GET https://127.0.0.1:51962/apis/company.com/v1/applysets/my-set 200 OK in 1 milliseconds
error: failed to fetch ApplySet parent object "applysets.company.com/my-set": no kind "ApplySet" is registered for version "company.com/v1" in scheme "pkg/scheme/scheme.go:28"
```
I believe this is happening in this function, after the successful request returns:

https://github.com/kubernetes/kubernetes/blob/f69c1c47463ff70ad61adf6f38c4d5b7373e9d0a/staging/src/k8s.io/client-go/rest/request.go#L1291-L1294

Note the "which means it passes through the decoder". I'm not yet sure why this doesn't happen in the tests, and moreover why it doesn't happen during the main apply operation on a CR.

Reply:
It seems to work with the unstructuredclient:: https://github.com/kubernetes/kubernetes/commit/a9f020fd6dc81e8e8a519106e55202d292e39256 

I don't know why we have so many clients though, it seems like dynamic client should be the same as unstructured client...